torch/_dynamo/variables/misc.py

```diff
@@ -744,7 +802,17 @@ def call_method(
         from ..trace_rules import is_callable_allowed
         from .builder import wrap_fx_proxy
 
-        if name == "apply":
+        # There are two cases to handle the apply method of an autograd function:
+        # 1. If the autograd function is not vmapified:
+        #    - We can directly handle it by either treating it as allow_in_graph or
+        #      wrapping it as an AutogradFunctionApplyVariable HOP.
+        # 2. If the autograd function is vmapified, there are two types to consider within the same process:
+        #    - The vmapped autograd function (name starts with "Vmapped"):
+        #      - We treat it as allow_in_graph or wrap it as an AutogradFunctionApplyVariable HOP.
+        #    - The original autograd function (be called when functorch transforms are active):
+        #      - Since we already wrap the vmapped autograd function as an AutogradFunctionApplyVariable HOP,
+        #        and the vmapped autograd function calls the original autograd function, we simply inline them.
+        if name == "apply" and not torch._C._are_functorch_transforms_active():
```

Comment:
Er, what happens if functorch transforms are active but you have a regular autograd.Function? Something like:

```
def f(x, y):
    z = Foo.apply(y)
    return x * z

x = torch.randn(3)
y = torch.randn([])
vmap(f, (0, None))(x, y)
```

Reply:
It goes into the `else` branch down blow, which is inlining the `apply` method. This is because we only want `call_apply` on the vmapped autograd function and capture fwd/bwd graphs there. And during tracing vmapped autograd function, it triggers call to the original regular autograd function's apply method, which we should just inline it.